sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -61,55 +142,75 @@ impl BlobClient {
             .per_call_policies
             .push(storage_headers_policy);
 
-        let client = GeneratedBlobClient::new(
-            endpoint,
-            credential.clone(),
-            container_name.clone(),
-            blob_name.clone(),
-            Some(options),
-        )?;
-        Ok(Self {
-            endpoint: endpoint.parse()?,
-            client,
-        })
-    }
-
-    /// Returns a new instance of AppendBlobClient.
-    ///
-    /// # Arguments
-    ///
-    pub fn append_blob_client(&self) -> AppendBlobClient {
-        AppendBlobClient {
-            endpoint: self.client.endpoint.clone(),
-            client: self.client.get_append_blob_client(),
+        let mut url = Url::parse(endpoint)?;
+        if !url.scheme().starts_with("http") {
+            return Err(azure_core::Error::message(
+                azure_core::error::ErrorKind::Other,
+                format!("{url} must use http(s)"),
+            ));
         }
-    }
 
-    /// Returns a new instance of BlockBlobClient.
-    ///
-    /// # Arguments
-    ///
-    pub fn block_blob_client(&self) -> BlockBlobClient {
-        BlockBlobClient {
-            endpoint: self.client.endpoint.clone(),
-            client: self.client.get_block_blob_client(),
-        }
+        // In regular ctor, since struct now holds blob_url, we have to build the blob_url given endpoint + container_name + blob_name
+        url.path_segments_mut()
+            .expect("Cannot be base")
+            .extend([&container_name, &blob_name]);
+
+        let client =
+            GeneratedBlobClient::from_url_with_credential(url.as_str(), credential, Some(options))?;
```

Comment:
In this instance, the regular convenience `new()` will use `from_url_with_credential` underlying